packages/manager/src/features/Lish/Weblish.tsx

```diff
@@ -56,24 +62,78 @@ export class Weblish extends React.Component<Props, State> {
   connect() {
     const { weblish_url, ws_protocols } = this.props;
 
-    this.socket = new WebSocket(weblish_url, ws_protocols);
+    /* When this.socket != origSocket, the socket from this connect()
+     * call has been closed and possibly replaced by a new socket. */
+    const origSocket = new WebSocket(weblish_url, ws_protocols);
+    this.socket = origSocket;
+
+    this.lastMessage = '';
+    this.setState({error: ''});
 
     this.socket.addEventListener('open', () => {
       if (!this.mounted) {
         return;
       }
-      this.setState({ renderingLish: true }, () => this.renderTerminal());
+      this.setState({ renderingLish: true }, () => this.renderTerminal(origSocket));
+    });
+
+    this.socket.addEventListener('close', (evt) => {
+      if (this.socket !== origSocket) {
+        return;
+      }
+      this.socket = null;
+      this.terminal?.dispose();
+      this.setState({ renderingLish: false });
+      if (!this.mounted) {
+        return;
+      }
+
+      let parsed = null;
+      if (evt?.reason) {
+        try {
+          parsed = JSON.parse(evt.reason);
+        } catch {
+        }
+      }
+      if (parsed === null) {
+        try {
+          parsed = JSON.parse(this.lastMessage);
+        } catch {
+        }
+      }
```

Comment:
And these try catches
```suggestion
      if (evt?.reason) {
        try {
          parsed = JSON.parse(evt.reason);
          if (parsed === null) {
            parsed = JSON.parse(this.lastMessage);
          }
        } catch {}
      }
```

Reply:
I made a similar change. (Unlike the original code, the suggestion above does not try to parse `this.lastMessage` when `evt?.reason` is not set. The update that I submitted _also_ moves the check for `evt?.reason` inside the try/catch block which maintains the intended logic while reducing the total amount of code.)